src/platforms/common/stm32/rtt_if.c

```diff
@@ -121,8 +121,11 @@ uint32_t rtt_write(const char *buf, uint32_t len)
 	if (len != 0 && usbdev && usb_get_config() && gdb_serial_get_dtr()) {
 		for (uint32_t p = 0; p < len; p += CDCACM_PACKET_SIZE) {
 			uint32_t plen = MIN(CDCACM_PACKET_SIZE, len - p);
-			while (usbd_ep_write_packet(usbdev, CDCACM_UART_ENDPOINT, buf + p, plen) <= 0)
-				continue;
+			uint32_t start_ms = platform_time_ms();
+			while (usbd_ep_write_packet(usbdev, CDCACM_UART_ENDPOINT, buf + p, plen) <= 0) {
+				if (platform_time_ms() - start_ms >= 10)
```

Comment:
Does it make sense to wait a little longer - say, 25 or 50ms - in case the host happens to be busy and a little slow? We'd be concerned with 10ms not being long enough and causing the possibility of dropping data wrongly just because things got a little busy on the USB link.